lib/streamlit/MediaFileManager.py

```diff
@@ -138,6 +155,7 @@ def clear_session_files(self, session_id=None):
         LOGGER.debug(
             "Sessions still active: %r", self._files_by_session_and_coord.keys()
         )
+        self._del_expired_files()
```

Comment:
This means that files from one session will linger until some other session happens to start or end, and that even happens to take place KEEP_DELAY secs later. To improve this, how about running `_del_expired_files` from a timer?

The only wrinkle with this idea is that one of the places where `clear_session_files` is called is when the script is about to run, so we can't simply add a `time.sleep(KEEP_DELAY)` here. So we have to schedule the run via Tornado's IOLoop instead. But since this function is always called from the actual Script thread, we can't just use `tornado.ioloop.IOLoop.current()` to get the current IOLoop. So we need to do a tiny bit of piping:

```py
# In Server.py __init__ method
    media_file_manager.set_ioloop(ioloop = self._ioloop)

# In MediaFileManager
    def __init__(self):
        # (..._)
        self._ioloop = None

    def set_ioloop(self, ioloop):
        self._ioloop = ioloop

    def clear_session_files(...):
        # (...)
        self._ioloop.call_later(KEEP_DELAY_SEC, self._del_expired_files)
```

Reply:
Yeah this seems to work.  I've had to adjust KEEP_DELAY_SEC to be longer (4 or 5, more testing required).

A lot of tests need to be adjusted now, due to the change in Server.py...